recipes/fuse_mnt.rb

```diff
@@ -0,0 +1,160 @@
+include_recipe "hops::default"
+
+# Intall the needed packages
+case node['platform_family']
+when 'debian'
+  package 'fuse'
+  package 'libfuse-dev'
+when 'rhel'
+  package 'fuse'
+  package 'fuse-libs'
+end
+
+# Allow non root users to mount file systems 
+bash "fuseconfig" do
+  user "root" 
+  code <<-EOH
+    echo "user_allow_other" >>  /etc/fuse.conf
+  EOH
+  not_if "grep '^user_allow_other' /etc/fuse.conf"
+end
+
+directory node['hops']['fuse']['staging_folder'] do
+  action :create
+  owner node['hops']['hdfs']['user']
+  group node['hops']['group']
+  mode "0750"
+  recursive true
+end
+
+directory node['hops']['fuse']['mount_point'] do
+  action :create
+  owner node['hops']['hdfs']['user']
+  group node['hops']['group']
+  mode "0700"
+  recursive true
+end
+
+# download fuse mount bin
+fuse_mount_bin_url = node['hops']['fuse']['dist_url']
+bin_name = File.basename(fuse_mount_bin_url)
+fuse_mount_bin = "#{node['hops']['sbin_dir']}/#{bin_name}"
+
+file fuse_mount_bin do
```

Comment:
During upgrade, the binary will suddenly disappear while the filesystem is still mounted. I would follow the same practise we do everywhere else, a version-less symlink pointing to the binary with the version. Deleting it might be a bit unpredictable.